docs/KernelFeatures/UsingSecureVault.md

```diff
@@ -0,0 +1,108 @@
+# Secure Vault
+> For the full list of capabilities available in this kernel version, see the **features** section in the [root README.md file](../../README.md#key-features-and-tools). 
+
+WSO2 Carbon is shipped with a Secure Vault implementation. This allows you to store encrypted passwords that are mapped to aliases, i.e., you can use the aliases instead of the actual passwords in your configuration files for better security. For example, some configurations require the admin username and password. If the admin user password is "admin", you could use the alias `UserManager.AdminUser.Password` in your configuration file. You would then map that alias to the actual password "admin". At runtime, the product will look up this alias in the secure vault and SecureVault will return the decrypted password.
+
+There are three files that are needed by the SecureVault:
+
+1. Secure-vault.yaml: 
+    Configurations that are needed to configure the SecureVault is given in this file
```

Comment:
*Configurations that are required for configuring the SecureVault are given in this file: